// react-native-autocomplete-input/dist/index.js
import React from 'react';
import { FlatList, Platform, StyleSheet, Text, TextInput, View } from 'react-native';
function defaultKeyExtractor(_, index) {
    return `key-${index}`;
}
function defaultRenderItems({ item }) {
    return <Text>{String(item)}</Text>;
}
export const AutocompleteInput = React.forwardRef(function AutocompleteInputComponent(props, ref) {
    const { data, containerStyle, hideResults, inputContainerStyle, listContainerStyle, onShowResults, onStartShouldSetResponderCapture = () => false, renderResultList: ResultList = FlatList, renderTextInput: AutoCompleteTextInput = TextInput, flatListProps, style, ...textInputProps } = props;
    function renderResultList() {
        const listProps = {
            data,
            renderItem: defaultRenderItems,
            keyExtractor: defaultKeyExtractor,
            ...flatListProps,
            style: [styles.list, flatListProps?.style],
        };
        return <ResultList {...listProps}/>;
    }
    function renderTextInput() {
        const renderProps = {
            ...textInputProps,
            style: [styles.input, style],
            ref,
        };
        return <AutoCompleteTextInput {...renderProps}/>;
    }
    const showResults = data.length > 0;
    onShowResults?.(showResults);
    return (<View style={[styles.container, containerStyle]}>
      <View style={[styles.inputContainer, inputContainerStyle]}>{renderTextInput()}</View>
      {!hideResults && (<View style={listContainerStyle} onStartShouldSetResponderCapture={onStartShouldSetResponderCapture}>
          {showResults && renderResultList()}
        </View>)}
    </View>);
});
const border = {
    borderColor: '#b9b9b9',
    borderRadius: 1,
    borderWidth: 1,
};
const androidStyles = StyleSheet.create({
    container: {
        flex: 1,
    },
    input: {
        backgroundColor: 'white',
        height: 40,
        paddingLeft: 3,
    },
    inputContainer: {
        ...border,
        marginBottom: 0,
    },
    list: {
        ...border,
        backgroundColor: 'white',
        borderTopWidth: 0,
        margin: 10,
        marginTop: 0,
    },
});
const iosStyles = StyleSheet.create({
    container: {
        zIndex: 1,
    },
    inputContainer: {
        ...border,
    },
    input: {
        backgroundColor: 'white',
        height: 40,
        paddingLeft: 3,
    },
    list: {
        ...border,
        backgroundColor: 'white',
        borderTopWidth: 0,
        left: 0,
        position: 'absolute',
        right: 0,
    },
});
const styles = StyleSheet.create({
    ...Platform.select({
        android: androidStyles,
        ios: iosStyles,
        default: iosStyles,
    }),
});
export default AutocompleteInput;
